frame/executive/src/lib.rs

```diff
@@ -355,6 +357,14 @@ where
 
 		// post-extrinsics book-keeping
 		<frame_system::Module<System>>::note_finished_extrinsics();
+
+		let weight =  <frame_system::Module<System>>::block_weight();
+		let max_weight =  <System::BlockWeights as frame_support::traits::Get<_>>::get().max_block;
+		let remaining_weight = max_weight.saturating_sub(weight.total());
+
+		<frame_system::Module<System> as OnIdle<System::BlockNumber>>::on_idle(block_number, remaining_weight);
+		<AllModules as OnIdle<System::BlockNumber>>::on_idle(block_number, remaining_weight);
```

Comment:
here we should write in BlockWeight storage the final weight reduced by on_idle.

(and same below)